graph/builder.go

```diff
@@ -675,45 +675,20 @@ func (b *Builder) handleNetworkUpdate(vb *ValidationBarrier,
 	defer b.wg.Done()
 	defer vb.CompleteJob()
 
-	// If this message has an existing dependency, then we'll wait until
-	// that has been fully validated before we proceed.
-	err := vb.WaitForDependants(update.msg)
-	if err != nil {
```

Comment:
shall we move this to the `discovery` package since it is only used there now?

Reply:
ah - i see there is still an instantiation of it here - but do we need it at all? 
afaict, it has no effect anymore since the only call we do on it is `CompleteJob` which only serves a purpose if other processes are taking tokens out of the semaphore as well. 

In fact, I actually think because of how it is used now: All calls to `CompleteJob` will actually hang until the builder shuts down. The only reason it shuts down successfully now is cause of the quit channel of the builder (and hence the validation barrier's quit channel) gets closed.

Reply:
Great catch